provider/google.go

```diff
@@ -263,14 +244,71 @@ func (p *googleProvider) submitChange(zone string, change *dns.Change) error {
 		return nil
 	}
 
-	_, err := p.changesClient.Create(p.project, zone, change).Do()
+	zones, err := p.Zones()
 	if err != nil {
 		return err
 	}
 
+	// separate into per-zone change sets to be passed to the API.
+	changes := separateChange(zones, change)
+
+	for z, c := range changes {
+		if _, err := p.changesClient.Create(p.project, z, c).Do(); err != nil {
+			return err
+		}
+	}
+
 	return nil
 }
 
+// separateChange separates a multi-zone change into a single change per zone.
+func separateChange(zones map[string]*dns.ManagedZone, change *dns.Change) map[string]*dns.Change {
+	changes := make(map[string]*dns.Change)
+
+	for _, z := range zones {
+		changes[z.Name] = &dns.Change{
+			Additions: []*dns.ResourceRecordSet{},
+			Deletions: []*dns.ResourceRecordSet{},
+		}
+	}
+
+	for _, a := range change.Additions {
+		if zone := suitableManagedZone(ensureTrailingDot(a.Name), zones); zone != nil {
+			changes[zone.Name].Additions = append(changes[zone.Name].Additions, a)
+		}
+	}
+
+	for _, d := range change.Deletions {
+		if zone := suitableManagedZone(ensureTrailingDot(d.Name), zones); zone != nil {
+			changes[zone.Name].Deletions = append(changes[zone.Name].Deletions, d)
+		}
+	}
+
+	// separating a change could lead to empty sub changes, remove them here.
+	for zone, change := range changes {
+		if len(change.Additions) == 0 && len(change.Deletions) == 0 {
+			delete(changes, zone)
+		}
+	}
+
+	return changes
+}
+
+// suitableManagedZone returns the most suitable zone for a given hostname and a set of zones.
+func suitableManagedZone(hostname string, zones map[string]*dns.ManagedZone) *dns.ManagedZone {
```

Comment:
same as above in aws it is called `suitableZone`, so maybe make it a method on googleprovider ?